tools/IceRpc.ProtocGen/ClientGenerator.cs

```diff
@@ -57,15 +57,17 @@ internal static string GenerateImplementation(ServiceDescriptor service)
             Invoker,
             ServiceAddress,
             ""{method.Name}"",
-            message.ToPipeReader(),
+            message.EncodeAsLengthPrefixedMessage(EncodeOptions?.PipeOptions ?? ProtobufEncodeOptions.Default.PipeOptions),
             payloadContinuation: null,
             async (response, request, cancellationToken) =>
```

Comment:
It seems this could be much simpler:
- call `Invoke.InvokeAsync<T>(...)`
- don't include a ResponseDecodeFunc param since it's always the same code

Reply:
With Slice, there is no special T that represents the result, hence we need this decode-response-func parameter.

Reply:
I would defer this to a follow-up PR.